services/frontend-service/pkg/cmd/server.go

```diff
@@ -114,8 +119,23 @@ func runServer(ctx context.Context) error {
 		grpc_zap.UnaryServerInterceptor(grpcServerLogger),
 	}
 
+	logger.FromContext(ctx).Info("config.tls_enabled: " + strconv.FormatBool(c.TlsEnabled) + "\n")
+
+	var cred credentials.TransportCredentials = insecure.NewCredentials()
+	if c.TlsEnabled {
+		systemRoots, err := x509.SystemCertPool()
+		if err != nil {
+			msg := "failed to read CA certificates"
+			logger.FromContext(ctx).Error(msg)
+			return fmt.Errorf(msg)
```

Comment:
this looks like it would log twice. Once here and once in the calling function that prints the error again.
-> Remove the log here.